nixos/modules/services/misc/gitea.nix

```diff
@@ -386,11 +386,11 @@ in
           HTTP_PORT = cfg.httpPort;
         })
         (mkIf cfg.ssh.enable {
-          DISABLE_SSH = false;
-          SSH_PORT = cfg.ssh.clonePort;
+          DISABLE_SSH = lib.mkDefault false;
+          SSH_PORT = lib.mkDefault cfg.ssh.clonePort;
         })
         (mkIf (!cfg.ssh.enable) {
-          DISABLE_SSH = true;
+          DISABLE_SSH = lib.mkDefault true;
```

Comment:
Why though? Isn't `ssh.enable` there to set a reasonable value here?

If somebody wants to leave the "supported route" of how this module is configured, `mkForce` is IMHO totally valid.

Reply:
When a value in `settings` takes its value from an option I think it is best to avoid `mkDefault` because it forces the user to realize they are overriding an option. Usually the idea behind this is that they are probably impacting the module in more than one way... though that is not the case here.

At the same time arguably `.ssh.enable`, `.ssh.clonePort`, `.cookieSecure`, `.disableRegistration` shouldn't even be options because they provide no additional value over setting a value in `.settings`. Personally I think we should remove these options as a solution to achieve what @SuperSandro2000 wants...

Failing to remove the options I would suggest @SuperSandro2000 use a blanket `mapAttrs (name: mkForce) { ... }` in his personal configuration instead of `mkDefault` on settings derived from options instead.

... but seriously - remove the options which provide no additional value? Any objections?

Reply:
Good idea!

Reply:
Pushed a first freeform settings change.